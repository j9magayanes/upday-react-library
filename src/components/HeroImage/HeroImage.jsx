import React from 'react';


export const HeroImage = ({ args}) => {
    return (
        <figure>
        <img className={args.class}src={args.src} alt={ args.shortTitle } ref={ ref => (args.imageRef = ref) }/>
        {
          args.photoCredits && args.photoCredits !== ' ' && <figcaption>{args.photoCredits}</figcaption>
        }
      </figure>
    )
}