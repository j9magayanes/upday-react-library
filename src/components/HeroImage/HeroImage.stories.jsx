import React from 'react';

import { HeroImage } from './HeroImage';


export default {
  title: 'Form/HeroImage',
  component:HeroImage,
};

const Template = (args) =><HeroImage {...args} />;


export const Default = Template.bind({});
Default.args = {
    src: "https://img-staging.yana.asideas.de/image/fetch/w_700,h_394,c_fill,f_webp,q_auto:eco/https%3A%2F%2Fcdn-images.rtp.pt%2Fnoticias%2Fimages%2Fdefault_noticias.png%3Fw%3D860%26q%3D90%26auto%3Dformat",
    alt: "test",
    class: "hero",
    args: {  class: "hero",}
};