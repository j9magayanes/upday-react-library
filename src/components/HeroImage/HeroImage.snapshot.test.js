import React from 'react';
import renderer from 'react-test-renderer';
import { HeroImage } from './HeroImage';

test('renders HeroImage component correctly', () => {
  const props = {
    args: {
      class: 'your-class',
      src: 'your-image-src.jpg',
      shortTitle: 'Short Title',
      imageRef: null, // Replace with actual value if needed
      photoCredits: 'Photo Credits',
      // Add other required properties as needed
    },
  };

  const component = renderer.create(<HeroImage {...props} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
