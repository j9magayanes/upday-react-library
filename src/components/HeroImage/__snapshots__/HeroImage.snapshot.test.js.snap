// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HeroImage component correctly 1`] = `
<figure>
  <img
    alt="Short Title"
    className="your-class"
    src="your-image-src.jpg"
  />
  <figcaption>
    Photo Credits
  </figcaption>
</figure>
`;
